Library/Homebrew/install.rb

```diff
@@ -28,20 +29,27 @@ def perform_build_from_source_checks(all_fatal: false)
       Diagnostic.checks(:build_from_source_checks, fatal: all_fatal)
     end
 
+    def check_prefix
+      if Hardware::CPU.intel? && HOMEBREW_PREFIX == HOMEBREW_MACOS_ARM_DEFAULT_PREFIX
+        odie "Cannot install in Homebrew on Intel processor in ARM default prefix (#{HOMEBREW_PREFIX})!"
+      elsif Hardware::CPU.arm? && HOMEBREW_PREFIX == HOMEBREW_DEFAULT_PREFIX
+        odie "Cannot install in Homebrew on ARM processor in Intel default prefix (#{HOMEBREW_PREFIX})!"
```

Comment:
How do existing DTK users migrate? This error will appear for everyone who has tested ARM so far so it perhaps would be useful to point them somewhere.

I'm not entirely sure what this restriction does to help us. We already discourage using non-default prefixes so I'm not sure how this is any different?

Reply:
> How do existing DTK users migrate? This error will appear for everyone who has tested ARM so far so it perhaps would be useful to point them somewhere.

@Bo98 They'll need to create and setup a new installation in a different location. We recommend one in the installation documentation. Should we do so here, too?

> I'm not entirely sure what this restriction does to help us.

It avoids having ARM installs and Intel installs mixing. We don't have a way of enforcing that otherwise as the `arch` only starts being written to the tab as-of this PR.

> We already discourage using non-default prefixes so I'm not sure how this is any different?

I think it's better to actively forbid it because it allows side-by-side installations. I can see the use-case for having two side-by-side installations on ARM until we have complete bottle coverage (which will take years).

Reply:
@Bo98 Have added migration instructions in 2378a0bbb.